reactor-core/src/main/java/reactor/core/publisher/FluxSourceMono.java

```diff
@@ -55,4 +55,9 @@ public void subscribe(CoreSubscriber<? super I> actual) {
 		source.subscribe(actual);
 	}
 
+	@Override
+	public Object scanUnsafe(Attr key) {
+		if (key == Attr.RUN_STYLE) return Attr.RunStyle.SYNC;
```

Comment:
Isn't it `UNKNOWN`? Or `source. scanUnsafe(key)`?

Reply:
`FluxSourceMono` is a final class, which is the reason why I decided to mark it as `SYNC` but you're right, `source.scanUnsafe(key)` might be better as the Mono source could possibly be `ASYNC`

Reply:
yeah it would actually make sense to use `Scannable.from(source).scanUnsafe(key)`, since this is a final wrapper of `Mono`. Note we expect most `Mono` to be `Scannable`, but it is not guaranteed. However `Scannable.from(foo)` will return a _default object_ instance that only answers with default values if the `foo` is not actually explicitly scannable.